### PR TITLE
Address DBE review comments for operators adjacent to @@variables 

### PR DESCRIPTION
**Pull request by robverschoor**
### Description

Address DBE review comments for https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/2217 

Signed-off-by: Rob Verschoor [rcv@amazon.com](mailto:rcv@amazon.com)

### Issues Resolved

Additional test cases for use cases inside  views and functions.

### Test Scenarios Covered ###
* **Use case based -** Yes

* **Boundary conditions -** N/A

* **Arbitrary inputs -** N/A

* **Negative test cases -** N/A

* **Minor version upgrade tests -** N/A

* **Major version upgrade tests -** N/A

* **Performance tests -** N/A

* **Tooling impact -** N/A

* **Client tests -** N/A

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp [hunk 1]
```diff
@@ -3219,21 +3187,22 @@ class tsqlMutator : public TSqlParserBaseListener
...
-	// NB: this is copied in tsqlBuilder
+  }
 	void exitConstant(TSqlParser::ConstantContext *ctx) override
 	{
 		// Check for floating-point number without exponent
 		handleFloatWithoutExponent(ctx);
 	}
-
-// NB: this is copied in tsqlBuilder
+	void exitPredicate(TSqlParser::PredicateContext *ctx) override
```
Comment (forestkeeper) [inline contrib/babelfishpg_tsql/src/tsqlIface.cpp:3201]: Why this function implement appeared twice ?

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp [hunk 2]
```diff
@@ -2313,40 +2314,6 @@ class tsqlBuilder : public tsqlCommonMutator
 		// TO-DO
 	}
 
-	void exitPredicate(TSqlParser::PredicateContext *ctx) override
```
Comment (forestkeeper) [inline contrib/babelfishpg_tsql/src/tsqlIface.cpp:2316]: Just for double check, the real implement is not changed ? I read one by one , and didn't see any difference in implement.

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp [hunk 3]
```diff
@@ -2369,26 +2336,27 @@ class tsqlBuilder : public tsqlCommonMutator
...
-	
-	// NB: this is copied in tsqlMutator
 	void exitConstant(TSqlParser::ConstantContext *ctx) override
 	{	
 		// Check for floating-point number without exponent
 		handleFloatWithoutExponent(ctx);
 	}
-  
-	// NB: this is copied in TsqlMutator
+	void exitPredicate(TSqlParser::PredicateContext *ctx) override
```
Comment (forestkeeper) [inline contrib/babelfishpg_tsql/src/tsqlIface.cpp:2355]: It's exactly the same compare to 3201 line

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp (hunk 2 above)
Comment (robverschoor) [reply contrib/babelfishpg_tsql/src/tsqlIface.cpp:2316]: Some statements are handled by tsqlBuilder (like top-level SELECT statements in a batch), others by tsqlMutator (like such statements in a procedure body). this is a consequence of the structure of this code has ended up. Therefore some things need to be executed in both classes. This is not new.

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp (hunk 1 above)
Comment (robverschoor) [reply contrib/babelfishpg_tsql/src/tsqlIface.cpp:3201]: see comment below

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp (hunk 3 above)
Comment (robverschoor) [reply contrib/babelfishpg_tsql/src/tsqlIface.cpp:2355]: see comment above

**Review by forestkeeper — APPROVED** [review]

**Pull request merged**